Source/NSPopover.m

```diff
@@ -34,101 +34,290 @@
 #import "AppKit/NSPopover.h"
 #import "AppKit/NSViewController.h"
 #import "AppKit/NSView.h"
-#import "AppKit/NSWindow.h"
+#import "AppKit/NSPanel.h"
+#import "AppKit/NSNibLoading.h"
+#import "AppKit/NSStoryboard.h"
+#import "AppKit/NSGraphics.h"
 
+// Popover private classes
+
+@interface GSPopoverView : NSView
+{
+  NSView *_contentView;
+}
+@end
+
+@implementation GSPopoverView
+
+- (instancetype) initWithFrame: (NSRect)frame
+{
+  self = [super initWithFrame: frame];
+  if (self)
+    {
+      [self setAutoresizingMask: NSViewWidthSizable | NSViewHeightSizable];
+    }
+  return self;
+}
+
+- (void) setContentView: (NSView *)view
+{
+  if (view != nil)
+    {
+      ASSIGN(_contentView, view);
```

Comment:
You should decide whether to store this in _contentView or to have it as a sub view. Doing both is inconsistent and will leak memory.

Reply:
Looking at the code further down in GSPopoverPanel you don't need any of this at all.

Reply:
I need the subview to draw the box around the view (as in GSToolTip.m).  Otherwise the popover looks a bit flat.   Which part do I not nee
![popover](https://user-images.githubusercontent.com/27943/135746416-27cc4ec9-f83e-4e11-af69-76f35374e0c2.png)
d?